File: THCS/2023_2024/ThanhHoa_2324/Cau1/CAU1.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

// Hàm chuyển đổi từ PropCase sang join_case
string convertPropToJoinCase(const string &varName) {
    string result;
    for (size_t i = 0; i < varName.size(); i++) {
        char ch = varName[i];
        if (isupper(ch) && i > 0) { // Nếu là chữ hoa và không phải ký tự đầu
            result += '_';          // Thêm dấu gạch dưới
        }
        result += tolower(ch);       // Thêm chữ cái (chuyển thành chữ thường)
    }
    return result;
}

int main() {
    // Đọc dữ liệu từ tệp CAU1.INP
    ifstream inputFile("CAU1.INP");
    string propCaseVar;
    if (inputFile.is_open()) {
        getline(inputFile, propCaseVar);
        inputFile.close();
    }

    // Chuyển đổi từ PropCase sang join_case
    string joinCaseVar = convertPropToJoinCase(propCaseVar);

    // Ghi kết quả vào tệp CAU1.OUT
    ofstream outputFile("CAU1.OUT");
    if (outputFile.is_open()) {
        outputFile << joinCaseVar;
        outputFile.close();
    }

    return 0;
}

File: THCS/2023_2024/ThanhHoa_2324/Cau1/CAU1.py
def convert_prop_to_join_case(var_name):
    result = []
    for i, ch in enumerate(var_name):
        if ch.isupper() and i > 0:  # Nếu là chữ cái viết hoa và không phải ký tự đầu
            result.append('_')
        result.append(ch.lower())  # Thêm chữ cái (chuyển thành chữ thường)
    return ''.join(result)

# Đọc dữ liệu từ tệp CAU1.INP
with open('CAU1.INP', 'r') as file:
    prop_case_var = file.read().strip()

# Chuyển đổi từ PropCase sang join_case
join_case_var = convert_prop_to_join_case(prop_case_var)

# Ghi kết quả vào tệp CAU1.OUT
with open('CAU1.OUT', 'w') as file:
    file.write(join_case_var)
